### ISSUE TITLE
iOS - Turning off auto registration still prompts for permission

### ISSUE THREAD

**Issue by cbartonnh** [CrossGeeks/PushNotificationPlugin#44]
After working in my own app for a while I decided to try to sample app and I am unable to get it working there either.

When calling PushNotificationManager.Initialize(options, false) in FinishedLaunching, I am still getting prompted for permission. Source in GitHub shows that it should not be calling RegisterForRemoteNotificaitons() but Nuget binary seems to do it.

I tried updating the sample to later plugin, XF, etc. and still not working.

Is there something else I need to do in order to defer registration until I call Register()?

**Comment by cbartonnh** [comment]
I went back and looked again and I missed the permission call that is being made if iOS version is 10 or above.

It would seem like this is in conflict with the auto-registration feature since I would expect that the reason you want to defer is so you can provide some additional UI to justify why you want the user to accept the OS permission prompt that is about to be presented.

**Comment by rdelrosario** [comment]
Ok will review it and fix it. Thanks for reporting this

**Comment by rdelrosario** [comment]
Fixed the issue in latest version 1.3.0. Included methods for registration/unregistration on demand if needed:
```cs
CrossPushNotification.Current.RegisterForPushNotifications();
CrossPushNotification.Current.UnegisterForPushNotifications();
```
